### PR TITLE
Improved the readability and fixed few java warnings

### PR DESCRIPTION
**Pull request by mrsrinivas**
### Description
Fixed the readability and fixed few java warnings

<hr>

This PR has:
- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

<hr>

##### Key changed/added classes in this PR
 * `sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/TimeArithmeticOperatorConversion.java`
 * `sql/src/main/java/org/apache/druid/sql/calcite/filtration/ConvertSelectorsToIns.java`

### CODE REVIEW

**Comment by mrsrinivas** [comment]
retest

**Comment by mrsrinivas** [comment]
@gianm thank you for the review. I fixed it.

File: sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidConvention.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public RelTraitDef getTraitDef()
   @Override
   public boolean satisfies(RelTrait trait)
   {
-    return trait == this;
+    return trait.equals(this);
```
Comment (jihoonson) [inline sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidConvention.java]: Do we need this change? This class is singleton and equals() is not implemented for this class.

Comment (gianm) [reply sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidConvention.java]: It's ok, IMO.

**Pull request merged**